common/include/pcl/point_types.h

```diff
@@ -350,563 +338,8 @@ namespace pcl
-
-    template <typename PointT>
-    using HasNoXYZ = std::enable_if_t<!has_xyz_v<PointT>, bool>;
-
-    /** Metafunction to check if a given point type has normal_x, normal_y, and
-      * normal_z fields. */
-    template <typename PointT>
-    struct has_normal : has_all_fields<PointT, boost::mpl::vector<pcl::fields::normal_x,
-                                                                  pcl::fields::normal_y,
-                                                                  pcl::fields::normal_z> >
-    { };
-
-    template <typename PointT>
-    constexpr auto has_normal_v = has_normal<PointT>::value;
-
-    template <typename PointT>
-    using HasNormal = std::enable_if_t<has_normal_v<PointT>, bool>;
-
-    template <typename PointT>
-    using HasNoNormal = std::enable_if_t<!has_normal_v<PointT>, bool>;
-
-    /** Metafunction to check if a given point type has curvature field. */
-    template <typename PointT>
-    struct has_curvature : has_field<PointT, pcl::fields::curvature>
-    { };
-
-    template <typename PointT>
-    constexpr auto has_curvature_v = has_curvature<PointT>::value;
-
-    template <typename PointT>
-    using HasCurvature = std::enable_if_t<has_curvature_v<PointT>, bool>;
-
-    template <typename PointT>
-    using HasNoCurvature = std::enable_if_t<!has_curvature_v<PointT>, bool>;
-
-    /** Metafunction to check if a given point type has intensity field. */
-    template <typename PointT>
-    struct has_intensity : has_field<PointT, pcl::fields::intensity>
-    { };
-
-    template <typename PointT>
-    constexpr auto has_intensity_v = has_intensity<PointT>::value;
-
-    template <typename PointT>
-    using HasIntensity = std::enable_if_t<has_intensity_v<PointT>, bool>;
-
-    template <typename PointT>
-    using HasNoIntensity = std::enable_if_t<!has_intensity_v<PointT>, bool>;
-
-    /** Metafunction to check if a given point type has either rgb or rgba field. */
-    template <typename PointT>
-    struct has_color : has_any_field<PointT, boost::mpl::vector<pcl::fields::rgb,
-                                                                pcl::fields::rgba> >
-    { };
-
-    template <typename PointT>
-    constexpr auto has_color_v = has_color<PointT>::value;
-
-    template <typename PointT>
-    using HasColor = std::enable_if_t<has_color_v<PointT>, bool>;
-
-    template <typename PointT>
-    using HasNoColor = std::enable_if_t<!has_color_v<PointT>, bool>;
-
-    /** Metafunction to check if a given point type has label field. */
-    template <typename PointT>
-    struct has_label : has_field<PointT, pcl::fields::label>
-    { };
-
-    template <typename PointT>
-    constexpr auto has_label_v = has_label<PointT>::value;
-
-    template <typename PointT>
-    using HasLabel = std::enable_if_t<has_label_v<PointT>, bool>;
-
-    template <typename PointT>
-    using HasNoLabel = std::enable_if_t<!has_label_v<PointT>, bool>;
-  }
-
 } // namespace pcl
+/** @} */
```

Comment:
What's this for?

Reply:
It was already there before, should be to group in doxygen iirc. I just left it where it was, not sure why it's counting an edit.